internal/dump/agentpolicies.go

```diff
@@ -0,0 +1,175 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package dump
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+
+	"github.com/elastic/elastic-package/internal/common"
+	"github.com/elastic/elastic-package/internal/kibana"
+)
+
+const AgentPoliciesDumpDir = "agent_policies"
+
+type AgentPoliciesDumper struct {
+	client *kibana.Client
+
+	policies []AgentPolicy
```

Comment:
Maybe we don't need to cache the policies here, this would simplify the logic. In the installed objects dumper we cache the objects to avoid querying objects multiple times when ones depends on others.
```suggestion
```

Reply:
Makes sense. At least now, the same agent policy is not being queried several times.